### ISSUE TITLE
Reorganize structure

### ISSUE THREAD

**Issue by tristen** [mapbox/gazetteer#51]
With the addition of locations specific to [one region of the world](https://github.com/mapbox/gazetteer/pull/50) does it make sense to organize the folder/file structure? Having a root `mapbox-streets` directory doesn't quite make sense to me. While these locations were initially created to evaluate style layers using [mapbox streets](https://docs.mapbox.com/vector-tiles/reference/mapbox-streets-v8/), these locations are relevant to other datasets and its less clear to someone using gazetteer why it's labelled that way.

@dasulit @ClareTrainor what do you think about this proposal:

Conceptually, anything that is specific to `mapbox-streets` should use the `tags` field to cite that.

### Files

- Rename `poi-label-hierarchy.json` 👉 `points-of-interest.json`. Make this file more genericized with locations that deal with heirarchy as `tag: heirarchy`.
- Rename `place-settlement-hierarchy-and-density.json` 👉 `place-labels.json`. Make this file more genericized with locations that deal with heirarchy or density as `tag: heirarchy`, `tag: density`.
- Rename `transit-labels.json` 👉 `transit.json`.Make this file more genericized.
- Merge `suburb-quarter-neighbourhood.json` with `place-labels.json`. Distinguish theme using tags.

### Folder structure
```
locations/
├── global/
│   ├── highway-shields.json
│   ├── natural-labels.json
│   ├── place-labels.json
│   ├── points-of-interest.json
│   ├── road-networks.json
│   └── transit.json
├── japan/
│   ├── natural-labels.json
│   ├── place-labels.json
│   ├── points-of-interest.json
│   ├── road-networks.json
│   └── transit.json
├── benchmarks/
│   ├── mapbox-streets.json
│   └── README.md // Documents intent of this directory
├── README.md // Documents rules around naming and folder structure.
```

**Comment by dasulit** [comment]
👍 on board with the suggestions outlined here! I can take this on.

**Comment by ClareTrainor** [comment]
Nice, this is a great idea Tristen!

**Comment by dasulit** [comment]
Closed by #52.
